### PR TITLE
Dev giogio

### PR DESCRIPTION
**Pull request by GiorgioAlbertoLucia**
fileManager added to read data from file. Clusterer is now updated and performs the entire clustering algorithm. A new constructor is required in Cluster.h

### CODE REVIEW

File: Clustering/src/clusterer.cxx [hunk 1]
```diff
@@ -35,47 +37,62 @@ void Clusterer::Clustering(const char *inputFile) {
...
           continue;
 
         bool isValid = false;
-        if ((std::abs(x[index] - x[indices[0]]) +
-             std::abs(y[index] - y[indices[0]])) <= 1)
-          isValid = true;
-        for (int j : next_indices) {
-          if (std::abs(x[index] - x[j]) == 1 && std::abs(y[index] - y[j]) == 1)
+        for (int j : nextIndices) {
+          if (std::abs(x[index] - x[j]) == 1 + std::abs(y[index] - y[j]) <= 1)
```
Comment (fchinu) [inline Clustering/src/clusterer.cxx]:
```suggestion
          if (std::abs(x[index] - x[j]) + std::abs(y[index] - y[j]) <= 1)
```

**Review by fchinu — CHANGES_REQUESTED** [review]
Hello @GiorgioAlbertoLucia! I have finished reviewing your PR, which looks good to me. I only have this minor comment, then we are ready to start clustering! Please make sure to fix your branch history for next time, since this PR also contains older commits (already merged), making it difficult to review

**Comment by fchinu** [comment]
Many thanks @GiorgioAlbertoLucia! PR #7 added the requirement for clang-formatting, can you run `clang-format -i fileneme` to the files that do not pass this check? Also I see that I cannot merge because `This branch is out-of-date with the base branch
`, I guess you can click on `update branch` in your branch? I think I can do it, but don't want to risk to ruin your work

**Comment by GiorgioAlbertoLucia** [comment]
Done

**Review by fchinu — APPROVED** [review]
Thank you!

**Pull request merged**